pkgs/applications/video/video-trimmer/default.nix

```diff
@@ -0,0 +1,77 @@
+{ lib
+, stdenv
+, fetchFromGitLab
+, meson
+, ninja
+, pkg-config
+, rustPlatform
+, glib
+, gtk4
+, libadwaita
+, blueprint-compiler
+, gst_all_1
+, ffmpeg-full
+, wrapGAppsHook4
+, appstream-glib
+, desktop-file-utils
+}:
+stdenv.mkDerivation rec {
+  pname = "video-trimmer";
+  version = "0.7.1";
+
+  src = fetchFromGitLab {
+    domain = "gitlab.gnome.org";
+    owner = "YaLTeR";
+    repo = "video-trimmer";
+    rev = "v${version}";
+    sha256 = "sha256-D7wjJkdqqjjwwYEUZnNr7hFQK59wfTnaCLXCy+SK8Jo=";
+  };
+
+  cargoDeps = rustPlatform.fetchCargoTarball {
+    inherit src;
+    name = "${pname}-${version}";
+    hash = "sha256-cB5dVrEbISvHrOb87uVZSkT694VKtPtyk+c1tYNCTp0=";
+  };
+
+  nativeBuildInputs = [
+    meson
+    ninja
+    pkg-config
+    blueprint-compiler
+    wrapGAppsHook4
+    appstream-glib
+    desktop-file-utils
+  ] ++ (with rustPlatform; [
+    rust.cargo
+    rust.rustc
+    cargoSetupHook
+  ]);
+
+  buildInputs = [
+    glib
+    gtk4
+    libadwaita
+    ffmpeg-full
+  ] ++ (with gst_all_1; [
+    gstreamer
+    gst-editing-services
+    gst-plugins-base
+    gst-plugins-good
+    gst-plugins-bad
+    gst-plugins-ugly
+    gst-libav
+  ]);
+
+  # delete unused postinstall.py line
+  patchPhase = ''
```

Comment:
Overriding `patchPhase` breaks `patches` field. Use `postPatch` instead. See <https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/patch-phase.md> for more info.

Reply:
Actually, the script is needed for [compiling GSettings schemas](https://gitlab.gnome.org/YaLTeR/video-trimmer/-/blob/fe64a108f1bf7b57d4bc7d7dc456034c30b0fdea/build-aux/meson/postinstall.py#L19). How can it work without it?

Reply:
Yes, it works. and nix log show: 
`Installing /build/source/data/icons/hicolor/scalable/apps/org.gnome.gitlab.YaLTeR.VideoTrimmer.svg to /nix/store/q0d8
Installing /build/source/data/icons/hicolor/symbolic/apps/org.gnome.gitlab.YaLTeR.VideoTrimmer-symbolic.svg to /nix/store/q0d8
Installing /build/source/build/data/org.gnome.gitlab.YaLTeR.VideoTrimmer.gschema.xml to /nix/store/q0d8`
when i try include `postinstall.py` and compile, it alway failed.

Reply:
This will just install the schemas uncompiled, which will break at runtime:

```ShellSession
$ exa -T result/share/gsettings-schemas/
result/share/gsettings-schemas
└── video-trimmer-0.7.1
   └── glib-2.0
      └── schemas
         └── org.gnome.gitlab.YaLTeR.VideoTrimmer.gschema.xml
```

You need to add dependencies listed by `postinstall.py` to `nativeBuildInputs`.